Permanent/Network/Layer/APIRequestDispatcher.swift

```diff
@@ -111,10 +138,68 @@ class APIRequestDispatcher: RequestDispatcherProtocol {
         let result = verify(data: fileUrl, urlResponse: urlResponse, error: error)
         switch result {
         case .success(let url):
-            DispatchQueue.main.async {
-                completion(OperationResult.file(url as? URL, urlResponse))
+
+            guard let tempLocalURL = url as? URL else {
+                return
             }
 
+            do {
+                let documentFolderURL = try FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: false)
+                let fileURL = documentFolderURL.appendingPathComponent("mydownload.jpeg")
+                try FileManager.default.copyItem(at: tempLocalURL, to: fileURL)
+
+                DispatchQueue.main.async {
+                    //completion(OperationResult.file(fileURL, urlResponse))
+                }
+
+            } catch {
+                print("errrorr", error)
+            }
+
+//            if let dataURL = try? Data(contentsOf: fileURL) {
```

Comment:
This will be handled once the `downloadTask` implementation will be discussed and fixed with Dorel.